tests/foreman/cli/test_user.py

```diff
@@ -587,3 +587,89 @@ def test_negative_personal_access_token_invalid_date(self, target_sat):
+
+    :steps:
+        1. Create an admin user and a non-admin user with "edit_users" and "register_hosts" permission.
+        2. Generate a token with admin user and register a host with it, it should be successful.
+        3. Invalidate the token and try to use the generated token again to register the host, it should fail.
+        4. Invalidate tokens for multiple users with "invalidate-multiple" command, it should invalidate all the tokens for provided users.
+        5. Repeat Steps 2,3 and 4 with non-admin user and it should work the same way.
+
+    :expectedresults: Tokens invalidated cannot be used for registration of hosts.
+
+    :CaseImportance: Critical
+
+    :Verifies: SAT-30385
+    """
+    admin_login = gen_string('alpha')
+    non_admin_login = gen_string('alpha')
+    password = gen_string('alpha')
+    login = admin_login
+    admin_user = module_target_sat.cli_factory.user(
+        {
+            'login': admin_login,
+            'password': password,
+            'admin': '1',
+            'organization-ids': module_org.id,
+        }
+    )
+    user = admin_user
+    # Non-Admin user with "edit_users" permission and "Register hosts" role
+    non_admin_user = module_target_sat.cli_factory.user(
+        {'login': non_admin_login, 'password': password, 'organization-ids': module_org.id}
+    )
+    if auth_type == 'non-admin':
+        login = non_admin_login
+        user = non_admin_user
+        role = module_target_sat.cli_factory.make_role({'organization-id': module_org.id})
+        module_target_sat.cli_factory.add_role_permissions(
+            role.id,
+            resource_permissions={'User': {'permissions': ['edit_users']}},
+        )
+        module_target_sat.cli.User.add_role({'id': non_admin_user['id'], 'role-id': role['id']})
+        register_role = module_target_sat.satellite.cli.Role.info({'name': 'Register hosts'})
+        module_target_sat.cli.User.add_role(
+            {'id': non_admin_user['id'], 'role-id': register_role['id']}
+        )
+
+    # Generate token and verify token invalidation
+    cmd = module_target_sat.cli.HostRegistration.with_user(login, password).generate_command(
+        options={
+            'activation-keys': module_activation_key.name,
+            'insecure': 'true',
+            'organization-id': module_org.id,
+        }
+    )
+    result = rhel_contenthost.execute(cmd.strip('\n'))
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+    # Invalidate JWTs for a single user
+    result = module_target_sat.cli.User.with_user(login, password).invalidate(
+        options={
+            'user-id': user['id'],
+        }
+    )
+    assert f'Successfully invalidated registration tokens for {login}' in result
+
+    rhel_contenthost.unregister()
+    # Re-register the host with invalidated token
+    result = rhel_contenthost.execute(cmd.strip('\n'))
+    assert result.status == 1
+    assert "ERROR: unauthorized" in result.stdout
```

Comment:
Can you use single quotes for consistency?

Reply:
updated.